pdsfile/pdsfile.py

```diff
@@ -8,688 +13,376 @@
-    'link' : ('_linkshelf-', '_links'),
-}
-
-################################################################################
-# PdsLogger support
-################################################################################
-
-LOGGER = pdslogger.NullLogger()         # Optional logger
-
-def set_logger(logger=None):
-    """Set the PdsLogger."""
-
-    global LOGGER
-
-    if not logger:
-        logger = pdslogger.NullLogger()
-
-    LOGGER = logger
-
-def set_easylogger():
-    """Log all messages directly to stdout."""
-    set_logger(pdslogger.EasyLogger())
-
-################################################################################
-# Filesystem vs. shelf files
-################################################################################
-
 _GLOB_CACHE_SIZE = 200
-_PATH_EXISTS_CACHE_SIZE = 200
-SHELVES_ONLY = False
-
-FS_IS_CASE_INSENSITIVE = True
-# The above just a guess until preload. Certain unexpected behavior can arise
-# when the file system is case-insensitive, such as glob.glob matching files
-# that it really shouldn't. We only do the extra work to fix this if we know,
-# or believe, that the file system is case-insensitive. Note also that when
-# SHELVES_ONLY is True, the file system is effectively case-sensitive because
-# we only use file names and paths found in the shelf files.
+PATH_EXISTS_CACHE_SIZE = 200
+FILE_BYTE_UNITS = ['bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
 
-def use_shelves_only(status=True):
-    """Call before preload(). Status=True to identify files based on their
-    presence in the infoshelf files first. Search the file system only if a
-    shelf is missing."""
+##########################################################################################
+# Support functions for pdsfile/__init__.py
+##########################################################################################
+def construct_category_list(voltypes):
+    category_list = []
+    for checksums in ('', 'checksums-'):
+        for archives in ('', 'archives-'):
+            for voltype in voltypes:
+                category_list.append(checksums + archives + voltype)
 
-    global SHELVES_ONLY
+    category_list.remove('checksums-documents')
+    category_list.remove('archives-documents')
+    category_list.remove('checksums-archives-documents')
 
-    SHELVES_ONLY = status
+    return category_list
 
-################################################################################
-# How to handle missing shelf files
-################################################################################
 
-SHELVES_REQUIRED = False
+def logical_path_from_abspath(abspath, cls):
+    """Return the logical path derived from an absolute path.
 
-def require_shelves(status=True):
-    """Call before preload(). Status=True to raise exceptions when shelf files
-    are missing or incomplete. Otherwise, missing shelf info is only logged as a
-    warning instead.
+    Keyword arguments:
```

Comment:
Also, since `use_shelves` is class-specific, shouldn't `require_shelves` be as well?

Reply:
Both `use_shelves_only` and `require_shelves` are class-specific, please see line 622-646 in `pdsfile/pdsfile.py`

Reply:
But the problem with this implementation is if you call `PdsFile.use_shelves_only` or `PdsFile.require_shelves` it will "work", but won't actually set the flag in any subclass, resulting in unexpected bugs. I think we need to prevent these functions from being called on the parent class. Like logging, they should be copied into both of the subclasses.

Reply:
Fixed, both functions are moved to Pds3File & Pds4File classes, the parent PdsFile class won't be able to call them.